python/hail/methods/misc.py

```diff
@@ -142,6 +142,17 @@ def require_row_key_variant(dataset, method):
                          "  Found:{}".format(method, ''.join(
             "\n    '{}': {}".format(k, str(dataset[k].dtype)) for k in dataset.row_key)))
 
+def require_row_key_variant_w_struct_locus(dataset, method):
+    if (list(dataset.row_key) != ['locus', 'alleles'] or
+            (not isinstance(dataset['locus'].dtype, tlocus) and
+                 not isinstance(dataset['locus'].dtype, tstruct(contig=tstr, position=tint))) or
```

Comment:
you can check against the locus/alleles struct here to make it simpler

Reply:
Not sure what you mean.

Reply:
`dataset.row_key.dtype = hl.dtype('struct{locus: struct{contig: str, position: int32}, alleles: array<str>}')`